File: CS/WindowsApplication3/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraPivotGrid;

namespace WindowsApplication3
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        List<Object> yearValues;
        private void Form1_Load(object sender, EventArgs e)
        {
            // TODO: This line of code loads data into the 'nwindDataSet.CustomerReports' table. You can move, or remove it, as needed.
            this.customerReportsTableAdapter.Fill(this.nwindDataSet.CustomerReports);
        }

        private void pivotGridControl1_CustomCellDisplayText(object sender, DevExpress.XtraPivotGrid.PivotCellDisplayTextEventArgs e)
        {
            PivotGridControl pivot = sender as PivotGridControl;
            if (e.DataField == fieldPercent)
            {
                List<object> rowValues = new List<object>();
                foreach (PivotGridField field in e.GetRowFields())
                    rowValues.Add(UpdateValue(field, e.RowIndex, e));
                List<object> columnValues = new List<object>();
                foreach (PivotGridField field in e.GetColumnFields())
                    columnValues.Add(UpdateValue(field, e.ColumnIndex, e));
                object previousValue = e.GetCellValue(columnValues.ToArray(), rowValues.ToArray(), e.DataField);
                if (previousValue == null)
                    e.DisplayText = string.Empty;
                else if (Convert.ToDecimal(previousValue) == 0)
                    e.DisplayText = "err";
                else                    
                    e.DisplayText = (Convert.ToDecimal(e.Value) / Convert.ToDecimal(previousValue) - 1).ToString("P");
            }

        }

        private List<Object> YearValues
        {
            get
            {
                if (yearValues == null)
                {

                    yearValues = new List<object>();
                    foreach (Object val in fieldOrderDate.GetUniqueValues())
                        yearValues.Add(val);
                    yearValues.Sort();
                }

                return yearValues;
            }
        }

        private object UpdateValue(PivotGridField field, int index, PivotCellDisplayTextEventArgs e)
        {
            Object value = e.GetFieldValue(field, index);
            if (field == fieldOrderDate)
            {
                int currentPosition = YearValues.IndexOf(value);
                if (currentPosition > 0)
                    value = YearValues[currentPosition - 1];
                else
                    value = "0000";
            }
            return value;
        }
    }
}

File: VB/WindowsApplication3/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraPivotGrid

Namespace WindowsApplication3
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private yearValues_Renamed As List(Of Object)
		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' TODO: This line of code loads data into the 'nwindDataSet.CustomerReports' table. You can move, or remove it, as needed.
			Me.customerReportsTableAdapter.Fill(Me.nwindDataSet.CustomerReports)
		End Sub

		Private Sub pivotGridControl1_CustomCellDisplayText(ByVal sender As Object, ByVal e As DevExpress.XtraPivotGrid.PivotCellDisplayTextEventArgs) Handles pivotGridControl1.CustomCellDisplayText
			Dim pivot As PivotGridControl = TryCast(sender, PivotGridControl)
            If e.DataField Is fieldPercent Then
                Dim rowValues As New List(Of Object)()
                For Each field As PivotGridField In e.GetRowFields()
                    rowValues.Add(UpdateValue(field, e.RowIndex, e))
                Next field
                Dim columnValues As New List(Of Object)()
                For Each field As PivotGridField In e.GetColumnFields()
                    columnValues.Add(UpdateValue(field, e.ColumnIndex, e))
                Next field
                Dim previousValue As Object = e.GetCellValue(columnValues.ToArray(), rowValues.ToArray(), e.DataField)
                If previousValue Is Nothing Then
                    e.DisplayText = String.Empty
                ElseIf Convert.ToDecimal(previousValue) = 0 Then
                    e.DisplayText = "err"
                Else
                    e.DisplayText = (Convert.ToDecimal(e.Value) / Convert.ToDecimal(previousValue) - 1).ToString("P")
                End If
            End If

		End Sub

		Private ReadOnly Property YearValues() As List(Of Object)
			Get
				If yearValues_Renamed Is Nothing Then

					yearValues_Renamed = New List(Of Object)()
					For Each val As Object In fieldOrderDate.GetUniqueValues()
						yearValues_Renamed.Add(val)
					Next val
					yearValues_Renamed.Sort()
				End If

				Return yearValues_Renamed
			End Get
		End Property

		Private Function UpdateValue(ByVal field As PivotGridField, ByVal index As Integer, ByVal e As PivotCellDisplayTextEventArgs) As Object
			Dim value As Object = e.GetFieldValue(field, index)
			If field Is fieldOrderDate Then
				Dim currentPosition As Integer = YearValues.IndexOf(value)
				If currentPosition > 0 Then
					value = YearValues(currentPosition - 1)
				Else
					value = "0000"
				End If
			End If
			Return value
		End Function
	End Class
End Namespace